.github/workflows/docker.yaml

```diff
@@ -0,0 +1,80 @@
+name: Docker
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "v*.*.*"
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      tags:
+        description: "Tags to build and push"
+        required: true
+        type: string
+
+permissions:
+  contents: read
+  packages: write
+
+jobs:
+  build-push:
+    runs-on: ubuntu-latest
+
+    env:
+      DOCKER_REGISTRY: ghcr.io
+      DOCKER_IMAGE: osmosis-labs/meshd
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.20"
+          check-latest: true
+
+      - name: Extract Docker metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.DOCKER_REGISTRY }}/${{ env.DOCKER_IMAGE }}
+          tags: |
```

Comment:
Do you use the tags for anything?

Reply:
for versioning. As more versions of the mesh-security-sdk comes, we should be able to use the tagged version